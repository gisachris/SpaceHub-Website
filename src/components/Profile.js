import { useSelector } from 'react-redux';

function Profile() {
  const rocketData = useSelector((state) => state.rockets.Data);
  const bookedRockets = rocketData.filter((rockets) => rockets.reserved === true);
  return (
    <div className="profile">
      <div className="missions">
        <h2>Missions</h2>
        <p>No missions have been added yet!</p>
      </div>
      <div className="rockets">
        <h2 className="rocketsHeader">Rockets</h2>
        {bookedRockets.length === 0 && (
          <p>No rockets have been reserved yet!</p>
        )}

        {bookedRockets.length > 0 && (
          <table className="holderTable">
            <tbody>
              {bookedRockets.map((rocket) => (
                <tr key={rocket.id} className="rocketrow">
                  <td className="rocketdata">{rocket.name}</td>
                </tr>
              ))}
            </tbody>
          </table>
        )}
      </div>
    </div>
  );
}

export default Profile;
