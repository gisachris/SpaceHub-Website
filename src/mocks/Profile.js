import PropTypes from 'prop-types';

function Profile({ missionData, rocketData }) {
  return (
    <div className="profile">
      <div className="missions">
        {missionData.map((data) => (
          <ul key={data.id}>
            <li>{data.name}</li>
            <li>{data.desc}</li>
            <li>{data.join ? 'Joined' : 'Join'}</li>
          </ul>
        ))}
      </div>
      <div className="rockets">
        {rocketData.map((data) => (
          <ul key={data.id}>
            <li>{data.name}</li>
            <li>{data.desc}</li>
            <li>{data.reserved ? 'reserved' : 'Reserve'}</li>
          </ul>
        ))}
      </div>
    </div>
  );
}

export default Profile;

Profile.propTypes = {
  missionData: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      desc: PropTypes.string.isRequired,
      join: PropTypes.bool.isRequired,
    }),
  ).isRequired,
  rocketData: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      reserved: PropTypes.bool.isRequired,
    }),
  ).isRequired,
};
