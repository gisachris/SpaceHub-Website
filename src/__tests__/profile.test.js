/*eslint-disable*/
import {render} from '@testing-library/react';
import Profile from '../mocks/Profile';

describe('tests for Profile component', () => {
  test('renders correctly', () => {
    const missionData = [
      {
        id: 1,
        name: 'name 1',
        desc: 'desc 1',
        join: false,
      },
    ];
    const rocketData = [
      {
        id: 1,
        name: 'name 1',
        desc: 'desc 1',
        false: false,
      },
    ];

    const {container} = render(
      <Profile rocketData={rocketData} missionData={missionData} />
    );
    expect(container).toMatchSnapshot();
  });
});
