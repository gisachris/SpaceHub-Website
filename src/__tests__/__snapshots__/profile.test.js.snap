// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Profile component renders correctly 1`] = `
<div>
  <div
    class="profile"
  >
    <div
      class="missions"
    >
      <ul>
        <li>
          name 1
        </li>
        <li>
          desc 1
        </li>
        <li>
          Join
        </li>
      </ul>
    </div>
    <div
      class="rockets"
    >
      <ul>
        <li>
          name 1
        </li>
        <li>
          desc 1
        </li>
        <li>
          Reserve
        </li>
      </ul>
    </div>
  </div>
</div>
`;
